### ISSUE TITLE
Built-in functions(condition/action) no longer suggests arguments in some circumstances

### ISSUE THREAD

**Issue by Chromowolf** [zuhanit/epscript-lsp#28]
When the code is short, the `CreateUnit` would suggest arguments like this:
![image](https://user-images.githubusercontent.com/37732663/196232501-506b9f20-9c16-49eb-ba93-b02835f096b1.png)

But in the following example, it doesn't suggest arguments:
![image](https://user-images.githubusercontent.com/37732663/196232982-60b0b263-57ee-483d-87e0-8def257e1499.png)

FYI the full code is:
```js
function TrainSingleUnit(epd, unit: TrgUnit) {
    if (epd <= EPD(0x59CCA8) - 1) {
        println("invalid EPD: {}", epd);
        return -1;
    }
    // defect: no check Addons
    const u = EPDCUnitMap(epd);

    // check resources
    const q2, r2 = div(unit, 2);
    RawTrigger(conditions=r2.Exactly(1), actions=r2.SetNumber(2));
    const mineralCost = wread_epd(EPD(0x663888) + q2, r2);
    const gasCost = wread_epd(EPD(0x65FD00) + q2, r2);
    const player = u.playerID;
    if (!(
        Accumulate(player, AtLeast, mineralCost, Ore)
        && Accumulate(player, AtLeast, gasCost, Gas)
    )) {
        println("Not enough resource");
        return -1;
    }

    // check supply
    var unitSupply;
    const q4, r4 = div(unit, 4);
    switch(unit) {
      case $U("Zerg Zergling"), $U("Zerg Scourge"):
        unitSupply = 2;
        break;
      default:
        unitSupply = bread_epd(EPD(0x663CE8) + q4, r4);
    }
    const flag = bread_epd(EPD(0x6637A0) + q4, r4);  // specific race belong to(Group Flags), not user's race
    const CheckSupplyAvailable = Memory(0x582144, AtLeast, 0);
    const CheckSupplyMax = Memory(0x5821A4, AtLeast, 0);
    var flagBit, supplyUsed = 1, EPD(0x582174) + player;
    while (!flag.ExactlyX(0, 1 + 2 + 4)) {
        if (flag.ExactlyX(0, flagBit)) continue;
        const supply = unitSupply + dwread_epd(supplyUsed);
        const op1 =
            list(flag, Subtract, flagBit),
            list(EPD(CheckSupplyAvailable) + 2, SetTo, supply),
            list(EPD(CheckSupplyMax) + 2, SetTo, supply);
        SeqCompute(op1);
        if (!(CheckSupplyAvailable && CheckSupplyMax)) {
            println("Supply! {},flag={}, flagBit={}", supply,flag, flagBit);
            return -1;
        }
        EUDSetContinuePoint();
        const op2 =
            list(supplyUsed, Add, 36),
            list(EPD(CheckSupplyAvailable) + 1, Add, 36),
            list(EPD(CheckSupplyMax) + 1, Add, 36),
            // list(EPD(flag), Add, flagBit);
            list(flagBit, Add, flagBit);
        SeqCompute(op2);
    }

    //Train unit? Morph Larva?
    epdswitch (epd + 0x64/4, 255) {
      case $U("Zerg Larva"):  // larva
        u.orderState = 0;
        u.orderID = 42;
        u.buildQ1 = unit;  // invariant: Larva can't cancel and survive -> always use first slot
        SetResources(player, Subtract, mineralCost, Ore);
        SetResources(player, Subtract, gasCost, Gas);
        break;
      default:
        // Completed + GroundedBuilding + !unpowered
        if (!MemoryXEPD(epd + 0xDC/4, Exactly, 3, 0x403)) {
            println("Completed + GroundedBuilding + !unpowered");
            return -1;
        }

        // check build queue slot
        var offset, subp;
        const SetCP2BuildQ = SetMemory(0x6509B0, Add, 0);
        const Jump, OddBuildQ, EvenBuildQ = Forward(), Forward(), Forward();
        VProc(epd, list(
            offset.SetNumber(0x98/4),
            subp.SetNumber(0),
            SetMemory(0x6509B0, SetTo, 0xA4/4),
            epd.QueueAddTo(EPD(0x6509B0)),
            SetMemory(SetCP2BuildQ + 20, SetTo, 0x98/4 - 0xA4/4),
            SetNextPtr(Jump, OddBuildQ),
        ));
        RawTrigger(
            conditions=DeathsX(CurrentPlayer, AtLeast, 1, 0, 4),
            actions=SetNextPtr(Jump, EvenBuildQ)
        );
        RawTrigger(
            conditions=DeathsX(CurrentPlayer, AtLeast, 1, 0, 2),
            actions=list(SetMemory(SetCP2BuildQ + 20, Add, 2), offset.AddNumber(2))
        );
        RawTrigger(
            conditions=DeathsX(CurrentPlayer, AtLeast, 1, 0, 1),
            actions=list(SetMemory(SetCP2BuildQ + 20, Add, 1), offset.AddNumber(1))
        );
        const OddDone, EvenDone, ResetCP = Forward(), Forward(), Forward();
        Jump.__lshift__(RawTrigger(
            nextptr=0,
            actions=list(
                SetCP2BuildQ,
                SetNextPtr(OddDone, ResetCP),
                SetNextPtr(EvenDone, OddBuildQ),
            )
        ));
        PushTriggerScope();
        const End = Forward();
        OddBuildQ.__lshift__(RawTrigger(
            conditions=DeathsX(CurrentPlayer, Exactly, 228, 0, 0xFFFF),
            actions=offset.AddNumber(6 << 16)
        ));
        DoActions(list(
            SetNextPtr(ResetCP, EvenBuildQ),
            offset.AddNumber(1 << 16),
        ));
        OddDone.__lshift__(RawTrigger(
            nextptr=0,
            conditions=offset.AtLeast(5 << 16),
            actions=SetNextPtr(OddDone, End)
        ));
        ResetCP.__lshift__(RawTrigger(
            nextptr=0,
            conditions=offset.AtLeastX(0xA0/4, 0xFFFF),
            actions=list(
                SetNextPtr(ResetCP, OddBuildQ),
                SetMemory(0x6509B0, Subtract, 2),
                offset.SubtractNumberX(2, 0xFFFF),
            )
        ));
        EvenBuildQ.__lshift__(RawTrigger(
            conditions=DeathsX(CurrentPlayer, Exactly, 228 << 16, 0, 0xFFFF0000),
            actions=list(
                subp.SetNumber(2),
                offset.AddNumber(6 << 16),
            )
        ));
        DoActions(list(
            offset.AddNumber(1 << 16),
            SetMemory(0x6509B0, Add, 1),
        ));
        EvenDone.__lshift__(RawTrigger(
            nextptr=0,
            conditions=offset.AtLeast(5 << 16),
            actions=SetNextPtr(EvenDone, End)
        ));
        PopTriggerScope();
        End.__lshift__(setcurpl2cpcache());

        if (offset < 6 << 16) return -1;
        DoActions(offset.SetNumberX(0, 0xFFFF0000));

        wwrite_epd(epd + offset, subp, unit);
        u.secondaryOrderID = 38;  // Train(Normal)
        SetResources(player, Subtract, mineralCost, Ore);
        SetResources(player, Subtract, gasCost, Gas);
    }
    return 0;
}

var builder_ptr, builder_epd;

function onPluginStart() {
    setcurpl(0);
    GetGlobalStringBuffer();
    // AttackGround("loc1", "Anywhere", "Terran Marine", "Zerg Ultralisk", P1);
    builder_ptr, builder_epd = cunitepdread_epd(EPD(0x628438));
    CreateUnit(1, "Terran Barracks", "Anywhere", P1);
    CreateUnit(1, "Terran Command Center", "loc1", P1);
    SetResources(P1, SetTo, 1000, Ore);
    // TrainSingleUnit(builder_epd, "Terran Marine");
}

```

**Comment by zuhanit** [comment]
~~Current server can't parse multiple variable declaration. It doesn't included in `epScriptParser.g4`~~. Also cannot tokenize `epdswitch`. I'll fix it. Thank you!

__Edit__: Can parse, but cannot deal correctly.

**Comment by zuhanit** [comment]
Fixed in #30.
